client/me/purchases/manage-purchase/index.jsx

```diff
@@ -295,26 +326,57 @@ class ManagePurchase extends Component {
 		} );
 	};
 
+	getUpgradeUrl() {
+		const { purchase, siteId } = this.props;
+
+		const isUpgradeableBackupProduct = JETPACK_BACKUP_T1_PRODUCTS.includes(
+			purchase.productSlug ?? purchase.product_slug
+		);
+		const isUpgradeableSecurityPlan = JETPACK_SECURITY_T1_PLANS.includes(
+			purchase.productSlug ?? purchase.product_slug
+		);
+
+		if ( isUpgradeableBackupProduct || isUpgradeableSecurityPlan ) {
+			return `/plans/storage/${ siteId }/`;
+		}
+
+		return `/plans/${ siteId }/`;
+	}
+
 	renderUpgradeNavItem() {
-		const { purchase, translate, siteId } = this.props;
-		const buttonText = isExpired( purchase )
-			? translate( 'Pick Another Plan' )
-			: translate( 'Upgrade Plan' );
-
-		if (
-			! isPlan( purchase ) ||
-			isEcommerce( purchase ) ||
-			isComplete( purchase ) ||
-			isP2Plus( purchase )
-		) {
+		const { purchase, translate } = this.props;
+
+		const isUpgradeablePlan =
+			isPlan( purchase ) &&
+			! isEcommerce( purchase ) &&
+			! isComplete( purchase ) &&
+			! isP2Plus( purchase );
+
+		const isUpgradeableBackupProduct = JETPACK_BACKUP_T1_PRODUCTS.includes(
+			purchase.productSlug ?? purchase.product_slug
```

Comment:
This is not a big deal, but do we really not know what properties `purchase` will have? It looks like it always comes from `getByPurchaseId`, which returns a `Purchase` object with the `productSlug` property. Is there any case where it could be `product_slug`?

(Same comment for `isUpgradeableBackupProduct`, `isUpgradeableSecurityPlan` above.)

Reply:
Ahhh, I see now. I didn't look closely enough to see where `purchase` was coming from, but now since I can see it's guaranteed to be a `Purchase`, it doesn't make sense to check snake-case properties. Good catch!